tmt/steps/provision/local.py

```diff
@@ -47,15 +47,11 @@ def execute(self,
         environment.update(kwargs.pop('env', None) or {})
         environment.update(self.parent.plan.environment)
         if friendly_command is None:
-            if isinstance(command, (list, tuple)):
-                friendly_command = ' '.join(shlex.quote(s) for s in command)
-            else:
-                friendly_command = command
+            friendly_command = ' '.join(shlex.quote(s) for s in remote_command)
         # Run the command under the prepared environment
-        return self.run(command,
+        return self.run([' '.join(remote_command)],
```

Comment:
@happz, is this approach aligned with your #1378 vision? I mean joining the command list with spaces when passing it to `self.run()`? I thought that the intention is to make `self.run()` only accept a list of strings.

Reply:
Hmm, hard to say. I believe it could be seen as correct, `local` is passing a list of strings to `run()`, it's a list with a single string representing the argument for `shell=True`.

On the other hand, yep, it's slightly unexpected, and here I suppose it's up to us as developers to decide: with `shell=True`, should `run()` (or rather `_run()`) join the command into a single string & pass it to shell? Or should this be the responsibility of the caller? I'd probably pick the former option, leaving it to `run()`.

Any comments, ideas?

Reply:
Well, yes, for `shell=True` the requirement to have a list of strings (ok a list with a single string) is a little bit weird. A little bit more weird is what's actually hapenning under the hood. For example the following command

    tmt run plan --default provision -h local login --command 'echo one;echo two'

is `shlex.split` as an interactive command [here](https://github.com/teemtee/tmt/pull/1524/files#diff-d9422da1a3efb9209a7d4beb7d824f85a19e5abe91fb239f74806847d88db266R1162) into this:

    ['echo', 'one;echo', 'two']

in order to be again joined back with spaces here. I'm not sure this makes any sense. In a similar way, I'd say it does not make sense to `shlex.split` any more complex shell command line so that later it can be space-joined.

Reply:
```
$ tmt run plan --default provision -h local login --command 'echo one;echo two'
/var/tmp/tmt/run-093

/plans/default
    provision
        how: local
        distro: Fedora Linux 35 (Workstation Edition)
        login: Starting interactive shell
one
two
        login: Interactive shell finished
        summary: 1 guest provisioned
```